### PR TITLE
Ilrepack backport

### PR DESCRIPTION
**Pull request by jaffinito**
## Summary

Backports ILRepack from main branch to net35 branch.

## Details

- Pulls ilrepack out of home builder (this will be going away in my next PR) and adds it as a build event to the Core.csproj.
- Changed over to use `\licenses\` LICENSE.txt instead of the ones in Miscellaneous folder.  
- Added in `THIRD_PARTY_NOTICES.txt`
- Updated MSIinstaller, ArtifactBuilder to support new files
- NewRelic.Core output path updated to _build like everything else (prep for scripting)
- Removed buildCoreArtifactsForS3Deploy  from Homebuilder since its not used..

### CODE REVIEW

File: src/Agent/MsiInstaller/Installer/Product.wxs [hunk 1]
```diff
@@ -215,9 +215,12 @@ SPDX-License-Identifier: Apache-2.0
 		<!-- Main Product -->
 		<ComponentGroup Id="ProductComponents" Directory="NETAGENTFOLDER">
 			<!-- Text -->
-			<Component Id="LicenseComponent" Guid="{47114807-F20A-478D-9F4F-4B3D6CF195BE}">
-				<File Id="LicenseFile" Name="License.txt" KeyPath="yes" Source="$(var.AgentSourcePath)\Miscellaneous\License.txt" />
-			</Component>
+      <Component Id="LicenseComponent" Guid="{47114807-F20A-478D-9F4F-4B3D6CF195BE}">
+        <File Id="LicenseFile" Name="LICENSE.txt" KeyPath="yes" Source="$(var.RootSourcePath)\licenses\LICENSE.txt" />
```
Comment (lspangler584) [inline src/Agent/MsiInstaller/Installer/Product.wxs:219]: Can we delete the license.txt in Miscellaneous then?

File: src/Agent/NewRelic/Agent/Core/Core.csproj [hunk 2]
```diff
@@ -1,145 +1,188 @@
...
 
-  <Target Name="GenerateCode" BeforeTargets="CreateGitVersion" Condition="!Exists('$(ConfigurationCsPath)') Or !Exists('$(ExtensionCsPath)')">
-    <PropertyGroup>
-      <Xsd2CodePath>$(SolutionDir)BuildTools\Xsd2Code.exe</Xsd2CodePath>
-    </PropertyGroup>
-    <!-- <Exec Command="&quot;$(Xsd2CodePath)&quot; &quot;$(SolutionDir)NewRelic\Agent\Core\Config\Configuration.xsd&quot; NewRelic.Agent.Core.Config Configuration.cs /cl /ap /sc /xa" /> -->
-    <!-- <Exec Command="&quot;$(Xsd2CodePath)&quot; &quot;$(SolutionDir)NewRelic\Agent\Core\NewRelic.Agent.Core.Extension\Extension.xsd&quot; NewRelic.Agent.Core.Extension Extension.cs /cl /ap /sc /xa" /> -->
-  </Target>
+    <Reference Include="ICSharpCode.SharpZipLib">
+      <HintPath>$(SolutionDir)ExternalLibs\ICSharpCode.SharpZipLib.dll</HintPath>
```
Comment (lspangler584) [inline src/Agent/NewRelic/Agent/Core/Core.csproj:23]: ExternalLibs are in src\Agent

File: src/Agent/NewRelic/Agent/Core/Core.csproj [hunk 3]
```diff
@@ -1,145 +1,188 @@
...
-            <DesignTime>True</DesignTime>
-            <DependentUpon>Resources.resx</DependentUpon>
-        </Compile>
-
-        <!-- The following includes are required because these files are generated by the PreBuildEvent. They do not exist when building from a clean working folder. 
+  <ItemGroup>
+    <Compile Include="..\..\..\Shared\SharedAssemblyInfo.cs">
+      <Link>Properties\SharedAssemblyInfo.cs</Link>
+    </Compile>
+    <Compile Include="$(SolutionDir)Shared\SharedLog4NetRepository.cs">
```
Comment (lspangler584) [inline src/Agent/NewRelic/Agent/Core/Core.csproj:76]: Shared is in src\Agent

File: src/Agent/MsiInstaller/Installer/Product.wxs (hunk 1 above)
Comment (jaffinito) [reply src/Agent/MsiInstaller/Installer/Product.wxs:219]: Sure!

File: src/Agent/NewRelic/Agent/Core/Core.csproj (hunk 2 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Core.csproj:23]: This is correct, until the FullAgent.sln is moved.

File: src/Agent/NewRelic/Agent/Core/Core.csproj (hunk 3 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Core.csproj:76]: This is correct, until the FullAgent.sln is moved.

File: src/Agent/NewRelic/Agent/Core/Core.csproj (hunk 2 above)
Comment (lspangler584) [reply src/Agent/NewRelic/Agent/Core/Core.csproj:23]: understood, i was confused.

**Pull request merged**